optaplanner-examples/src/test/java/org/optaplanner/examples/conferencescheduling/solver/ConferenceSchedulingScoreRulesXlsxTest.java

```diff
@@ -0,0 +1,238 @@
+
+        final DateTimeFormatter DAY_FORMATTER
+                = DateTimeFormatter.ofPattern("E yyyy-MM-dd", Locale.ENGLISH);
+        final DateTimeFormatter TIME_FORMATTER
+                = DateTimeFormatter.ofPattern("HH:mm", Locale.ENGLISH);
+
+        public testConferenceSchedulingScoreRulesReader(XSSFWorkbook workbook) {
+            super(workbook);
+            this.numberOfSheets = workbook.getNumberOfSheets();
+            this.currentTestSheetIndex = workbook.getSheetIndex("Talks") + 1;
+            roomMap = currentSolution.getRoomList().stream().collect(
+                    Collectors.toMap(Room::getName, Function.identity()));
+            talkMap = currentSolution.getTalkList().stream().collect(
+                    Collectors.toMap(Talk::getCode, Function.identity()));
+            timeslotMap = currentSolution.getTimeslotList().stream().collect(
+                    Collectors.toMap(timeslot -> Pair.of(timeslot.getStartDateTime(), timeslot.getEndDateTime()),
+                                     Function.identity()));
+            this.columnIndexToDateMap = new HashMap<>(timeslotMap.size());
+            this.columnIndexToStartTimeMap = new HashMap<>(timeslotMap.size());
+            this.columnIndexToEndTimeMap = new HashMap<>(timeslotMap.size());
+        }
+
+        @Override
+        public ConferenceSolution read() {
+            return currentSolution;
+        }
+
+        public ConferenceSolution nextSolution() {
+            if (currentTestSheetIndex >= numberOfSheets) {
+                return null;
+            }
+            nextSheet(workbook.getSheetName(currentTestSheetIndex++));
+            currentSheetName = currentSheet.getSheetName();
+
+            ConferenceSolution nextSheetSolution = solutionCloner.cloneSolution(currentSolution);
+            talkMap = nextSheetSolution.getTalkList().stream().collect(
+                    Collectors.toMap(Talk::getCode, Function.identity()));
+
+            nextRow(false);
+            readHeaderCell("Constraint package");
+            currentPackage = nextStringCell().getStringCellValue();
+            nextRow(false);
+            readHeaderCell("Constraint name");
+            currentConstraint = nextStringCell().getStringCellValue();
+            nextRow(false);
+            readHeaderCell("Score");
+            String scoreAsString = nextStringCell().getStringCellValue();
+            if (!VALID_SCORE_PATTERN.matcher(scoreAsString).matches()) {
+                throw new IllegalStateException(currentPosition() + ": The score (" + scoreAsString
+                                                        + ") must match to the regular expression (" + VALID_SCORE_PATTERN + ").");
+            }
+            String[] scoresStringArray = scoreAsString.replaceAll(ONLY_DIGITS_AND_SPACES_REGEX, "").split(" +");
+            expectedHardScore = Integer.parseInt(scoresStringArray[0]);
+            expectedSoftScore = Integer.parseInt(scoresStringArray[1]);
```

Comment:
This code reinvents the wheel: it's doing score parsing itself. Remove those 2 regexes and just use HardSoftScore.parseScore(string).

Best to store the Score in a variable, instead of the hard/soft pieces separately. But if you do need them, you can easily get them with getHardScore().